### PR TITLE
chore(site): refactor logs and add stories

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Related to https://github.com/coder/coder/issues/12264

### CODE REVIEW

File: site/src/modules/resources/AgentLogLine.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          minWidth: `${maxNumber ? maxNumber.toString().length - 1 : 0}em`,
+        }}
+      >
+        {number}
+      </LogLinePrefix>
+      <LogLineSpace />
+      <span
+        dangerouslySetInnerHTML={{
+          __html: output,
+        }}
```
Comment (aslilac) [inline site/src/modules/resources/AgentLogLine.tsx:47]: why is this necessary? `dangerouslySetInnerHTML` should always be accompanied by an explaination

File: site/src/modules/resources/AgentLogLine.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return convert.toHtml(line.output.split(/\r/g).pop() as string);
+  }, [line.output]);
+
+  return (
+    <LogLine css={{ paddingLeft: 16 }} level={line.level} style={style}>
+      {sourceIcon}
+      <LogLinePrefix
+        css={styles.number}
+        style={{
+          minWidth: `${maxNumber ? maxNumber.toString().length - 1 : 0}em`,
```
Comment (aslilac) [inline site/src/modules/resources/AgentLogLine.tsx:38]: nit: `maxNumber` will only be falsey when it's `0`, and `0..toString().length - 1` is `0`, so the condition is a bit redundant.

it'd make more sense if `maxNumber` could be `undefined`, but that's not what the type says.

File: site/src/components/Logs/LogLine.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  );
+};
+
+export const LogLinePrefix: FC<HTMLAttributes<HTMLSpanElement>> = (props) => {
+  return <span css={styles.prefix} {...props} />;
+};
+
+export const LogLineSpace: FC = () => {
+  return <span css={styles.space} />;
+};
```
Comment (aslilac) [inline site/src/components/Logs/LogLine.tsx:35]: I'm not crazy about this component. I feel like there's gotta be a better way to add this padding/gap. A non-semantic empty span is a code smell to me.

File: site/src/components/Logs/LogLine.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  line: (theme) => ({
+    wordBreak: "break-all",
+    display: "flex",
+    alignItems: "center",
+    fontSize: 13,
+    color: theme.palette.text.primary,
+    fontFamily: MONOSPACE_FONT_FAMILY,
+    height: "auto",
+    // Whitespace is significant in terminal output for alignment
+    whiteSpace: "pre",
```
Comment (aslilac) [inline site/src/components/Logs/LogLine.tsx:47]: ...so why not use a `<pre>`? :)

File: site/src/modules/resources/AgentLogLine.tsx (hunk 1 above)
Comment (aslilac) [reply site/src/modules/resources/AgentLogLine.tsx:41]: Can we just add a comment above this that says "`output` contains HTML to represent ANSI-code formatting" or something then?

Comment (BrunoQuaresma) [reply site/src/modules/resources/AgentLogLine.tsx:41]: Sure thing!

Comment (BrunoQuaresma) [reply site/src/modules/resources/AgentLogLine.tsx:42]: Done!

**Review by aslilac — APPROVED** [review]
